backend/src/nodes/image_nodes.py

```diff
@@ -155,22 +155,32 @@ def run(
         img = (np.clip(img, 0, 1) * 255).round().astype("uint8")
 
         os.makedirs(base_directory, exist_ok=True)
-
-        # Write image with opencv if path is ascii, since imwrite doesn't support unicode
-        # This saves us from having to keep the image buffer in memory, if possible
-        if full_path.isascii():
-            status = cv2.imwrite(full_path, img)
+        # Any image not supported by cv2, will be handled by pillow.
+        if extension not in ["png", "jpg", "gif", "tiff", "webp"]:
+            status = 1  # spoof
+            if img.shape[2] == 4:
+                img = cv2.cvtColor(img, cv2.COLOR_BGRA2RGBA)
+            else:
+                img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
```

Comment:
Chainner also supports and creates single-channel images, so `img.shape` might only have 2 elements. 

1. Use `get_h_w_c(img)[2]` to reliably get the number of channels.
2. Add support for single-channel images.

We should probably also make it explicit that only single-channel, RGB, and RGBA images are supported for PIL. Maybe like this:

```py
c = get_h_w_c(img)[2]
if c == 1:
    img = ??? # idk how to do this in PIL
elif c == 3:
    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
elif c == 4:
    img = cv2.cvtColor(img, cv2.COLOR_BGRA2RGBA)
else:
    assert False, f"Unsupported number of channels. Saving .{extension} images is only supported for grayscale, RGB, and RGBA images."
```

Reply:
good point. I added the check. It now only supports 1,3,4 channels. fortunately grayscale 'L' channels just pass through in pillow so we don't need to do anything,